Dockerfile

```diff
@@ -0,0 +1,32 @@
+# base Docker image, remove "-alpine" if having trouble
+FROM node:10.5.0-alpine
+
+# added this step to prevent angular.json not found error
+WORKDIR /usr/src
+
+COPY . .
+
+# replace ng serve with ng serve --host 0.0.0.0 so that it serves outside
+# install jq for json edit
+RUN apk update \
+	&& apk add jq
+	
+# replace package.json
+RUN input_json="package.json" \
+	&& tmp_json="tmp.json" \
+	&& jq '.scripts.start = "ng serve --host 0.0.0.0"' $input_json > $tmp_json \
+	&& mv $tmp_json $input_json
+	
+# remove jq
+RUN apk del jq
+
+RUN npm install
+RUN npm audit fix

```

Comment:
In my opinion, we should not try to install packages inside of a docker as then we never know which version of the packages we run.

Reply:
I don't understand the reasoning. Aren't installed packages part of the environment? Also can't we just access terminal and running npm list to see packages' details?

Reply:
Running `npm audit fix` will update packages to a new version. Updating packages should be done in the regular package.json file outside the container, since version changes might bring incompatibilities (even if they're just bugfixes). Also you'll be able to version the changes to the package.json properly.